### ISSUE TITLE
Unable to add new user

### ISSUE THREAD

**Issue by tamirMoshiashvili** [thomasdarimont/embedded-spring-boot-keycloak-server#57]
Hey,
I cloned the project and tried to play with the embedded server.

### Error:
Through the console, I can create new realm, roles and groups just fine, but when I try to add new user, an error occurs - `updateReadOnlyAttributesRejectedMessage`.
Error log:
`2021-10-27 10:53:35.970  WARN 10896 --- [  XNIO-1 task-2] .u.v.ReadOnlyAttributeUnchangedValidator : Attempt to edit denied attribute '(?i:)' of user 'new user'`
I tried to debug the source of the problem in class `ReadOnlyAttributeUnchangedValidator` and it treats the field as readonly, even though I just try to add new user.

### Alternative ways to add new user:

- admin-console - **fails** as I mentioned above
- rest-api - **fails** with the same error
- import realm with users - **succeed**, but can't update user's details because of the same error above

I also tried to search for related **configuration** regarding read-only users and didn't manage to find something.

### Reproduce:

- Run the plain/custom embedded Keycloak server
- Enter admin console
- Create new realm
- Try to add user with some username

### Versions:
I checked versions `5.0.1`, `5.0.2`, `5.0.3` and the error occurs in all of them.
I checked against Keycloak 15.0.2 on docker - adding new user succeeds just fine.

**Comment by eRfO** [comment]
Hi,
the problem is related to the class com.github.thomasdarimont.keycloak.embedded.support.SpringBootConfigProvider.MapScope that in the getArray method return an empty array if the input key is not present in map. To correct the behaviour must be ruturn null ih the map does not contains the input key.

**Comment by tamirMoshiashvili** [comment]
Thanks for helping @eRfO.
Indeed, after changing `SpringBootConfigProvider.MapScope#getArray` to return null when `obj` is null, adding and updating users works as expected.

It seems that when returning empty array, the read-only attributes adds the wrong pattern - `(?i:)` (from the error-log above) which produces error when adding/updating user.
When returning null, there is no validation with that pattern.

I created [this PR](https://github.com/thomasdarimont/embedded-spring-boot-keycloak-server/pull/59) with the suggested change.

**Comment by bastoker** [comment]
As a workaround / quick fix it can help to put these props in the application yaml:
```
keycloak:
  userProfile:
    read-only-attributes: [""]
    admin-read-only-attributes: [""]
```

Since it seems these are the only Array properties that are interpreted in a strange way, when the array is returned but without any elements, by the new code introduced in Keycloak 14.
(version `5.0.2` of embedded-spring-boot-keycloak _does_ work fine when you change the keycloak version property to `13.0.1`)

**Comment by fzancan-SpazioCodice** [comment]
@bastoker Thanks, this saved my life.

**Comment by thomasdarimont** [comment]
This should be fixed now in Version 5.1.0 / 6.0.0 / 7.0.0
